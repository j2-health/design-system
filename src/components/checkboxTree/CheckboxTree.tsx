import React, { useState, useCallback } from 'react'
import { TreeNode } from './TreeNode'

export type CheckboxTreeDataNode = {
  key: string | number
  title: React.ReactNode
  children?: CheckboxTreeDataNode[]
  disabled?: boolean
  disableCheckbox?: boolean
}

const getChildrenKeys = (node: CheckboxTreeDataNode): (string | number)[] => {
  const keys: (string | number)[] = []
  const traverse = (currentNode: CheckboxTreeDataNode) => {
    if (currentNode.children) {
      currentNode.children.forEach((child) => {
        keys.push(child.key)
        traverse(child)
      })
    }
  }
  traverse(node)
  return keys
}

const getParentKey = (
  key: string | number,
  nodes: CheckboxTreeDataNode[]
): string | number | null => {
  for (const node of nodes) {
    if (node.children) {
      if (node.children.some((child) => child.key === key)) {
        return node.key
      }
      const parentKey = getParentKey(key, node.children)
      if (parentKey !== null) {
        return parentKey
      }
    }
  }
  return null
}

const findNode = (
  key: string | number,
  nodes: CheckboxTreeDataNode[]
): CheckboxTreeDataNode | null => {
  for (const node of nodes) {
    if (node.key === key) {
      return node
    }
    if (node.children) {
      const found = findNode(key, node.children)
      if (found) return found
    }
  }
  return null
}

export type CheckboxTreeProps = {
  treeData: CheckboxTreeDataNode[]
  defaultCheckedKeys?: (string | number)[]
  checkedKeys?: (string | number)[]
  onCheck?: (
    checkedKeys: (string | number)[],
    info: {
      checked: boolean
      checkedNodes: CheckboxTreeDataNode[]
      node: CheckboxTreeDataNode
    }
  ) => void
  disabled?: boolean
  className?: string
  style?: React.CSSProperties
}

export const CheckboxTree = ({
  treeData,
  defaultCheckedKeys = [],
  checkedKeys: controlledCheckedKeys,
  onCheck,
  disabled = false,
  className,
  style,
}: CheckboxTreeProps) => {
  const [internalCheckedKeys, setInternalCheckedKeys] = useState<
    Set<string | number>
  >(() => new Set(defaultCheckedKeys))

  const checkedKeys = controlledCheckedKeys
    ? new Set(controlledCheckedKeys)
    : internalCheckedKeys

  const handleCheck = useCallback(
    (key: string | number, checked: boolean, node: CheckboxTreeDataNode) => {
      const newCheckedKeys = new Set(checkedKeys)

      if (checked) {
        newCheckedKeys.add(key)
        // Check all children when parent is checked
        const childrenKeys = getChildrenKeys(node)
        childrenKeys.forEach((childKey) => newCheckedKeys.add(childKey))
      } else {
        newCheckedKeys.delete(key)
        // Uncheck all children when parent is unchecked
        const childrenKeys = getChildrenKeys(node)
        childrenKeys.forEach((childKey) => newCheckedKeys.delete(childKey))
      }

      // Update parent nodes based on children state
      let parentKey = getParentKey(key, treeData)
      while (parentKey !== null) {
        const parentNode = findNode(parentKey, treeData)
        if (parentNode && parentNode.children) {
          const allChildrenChecked = parentNode.children.every((child) =>
            newCheckedKeys.has(child.key)
          )

          if (allChildrenChecked) {
            newCheckedKeys.add(parentKey)
          } else {
            // If not all children are checked, uncheck the parent
            newCheckedKeys.delete(parentKey)
          }
        }
        parentKey = getParentKey(parentKey, treeData)
      }

      if (!controlledCheckedKeys) {
        setInternalCheckedKeys(newCheckedKeys)
      }

      if (onCheck) {
        const checkedNodes: CheckboxTreeDataNode[] = []
        const traverse = (nodes: CheckboxTreeDataNode[]) => {
          nodes.forEach((node) => {
            if (newCheckedKeys.has(node.key)) {
              checkedNodes.push(node)
            }
            if (node.children) {
              traverse(node.children)
            }
          })
        }
        traverse(treeData)

        onCheck(Array.from(newCheckedKeys), {
          checked,
          checkedNodes,
          node,
        })
      }
    },
    [checkedKeys, controlledCheckedKeys, onCheck, treeData]
  )

  return (
    <div className={className} style={style}>
      {treeData.map((node) => (
        <TreeNode
          key={node.key}
          node={node}
          level={0}
          checkedKeys={checkedKeys}
          onCheck={handleCheck}
          disabled={disabled}
        />
      ))}
    </div>
  )
}
