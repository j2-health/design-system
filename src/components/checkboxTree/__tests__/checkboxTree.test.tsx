import { render } from '@testing-library/react'
import { CheckboxTree } from '../CheckboxTree'

const simpleTreeData = [
  {
    key: '0',
    title: 'Documents',
    children: [
      {
        key: '0-0',
        title: 'Projects',
        children: [
          { key: '0-0-0', title: 'Project A' },
          { key: '0-0-1', title: 'Project B' },
        ],
      },
      {
        key: '0-1',
        title: 'Reports',
        children: [
          { key: '0-1-0', title: 'Annual Report' },
          { key: '0-1-1', title: 'Monthly Report' },
        ],
      },
    ],
  },
  {
    key: '1',
    title: 'Media',
    children: [
      { key: '1-0', title: 'Images' },
      { key: '1-1', title: 'Videos' },
    ],
  },
]

describe('CheckboxTree', () => {
  it('should render correctly', () => {
    const { container } = render(<CheckboxTree treeData={simpleTreeData} />)
    expect(container).toMatchSnapshot()
  })

  it('should render with some checked items', () => {
    const { container } = render(
      <CheckboxTree 
        treeData={simpleTreeData} 
        defaultCheckedKeys={['0-0-0', '1-0']} 
      />
    )
    expect(container).toMatchSnapshot()
  })

  it('should render disabled', () => {
    const { container } = render(
      <CheckboxTree 
        treeData={simpleTreeData} 
        disabled 
      />
    )
    expect(container).toMatchSnapshot()
  })
})