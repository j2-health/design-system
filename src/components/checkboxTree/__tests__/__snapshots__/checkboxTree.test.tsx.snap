// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxTree should render correctly 1`] = `
<div>
  <div>
    <div>
      <div
        class="flex items-center min-h-6 mb-1"
        style="padding-left: 0px;"
      >
        <label
          class="ant-checkbox-wrapper css-dev-only-do-not-override-os2fvf"
        >
          <span
            class="ant-checkbox ant-wave-target css-dev-only-do-not-override-os2fvf"
          >
            <input
              class="ant-checkbox-input"
              type="checkbox"
            />
            <span
              class="ant-checkbox-inner"
            />
          </span>
          <span
            class="ant-checkbox-label"
          >
            Documents
          </span>
        </label>
      </div>
      <div>
        <div>
          <div
            class="flex items-center min-h-6 mb-1"
            style="padding-left: 20px;"
          >
            <label
              class="ant-checkbox-wrapper css-dev-only-do-not-override-os2fvf"
            >
              <span
                class="ant-checkbox ant-wave-target css-dev-only-do-not-override-os2fvf"
              >
                <input
                  class="ant-checkbox-input"
                  type="checkbox"
                />
                <span
                  class="ant-checkbox-inner"
                />
              </span>
              <span
                class="ant-checkbox-label"
              >
                Projects
              </span>
            </label>
          </div>
          <div>
            <div>
              <div
                class="flex items-center min-h-6 mb-1"
                style="padding-left: 40px;"
              >
                <label
                  class="ant-checkbox-wrapper css-dev-only-do-not-override-os2fvf"
                >
                  <span
                    class="ant-checkbox ant-wave-target css-dev-only-do-not-override-os2fvf"
                  >
                    <input
                      class="ant-checkbox-input"
                      type="checkbox"
                    />
                    <span
                      class="ant-checkbox-inner"
                    />
                  </span>
                  <span
                    class="ant-checkbox-label"
                  >
                    Project A
                  </span>
                </label>
              </div>
            </div>
            <div>
              <div
                class="flex items-center min-h-6 mb-1"
                style="padding-left: 40px;"
              >
                <label
                  class="ant-checkbox-wrapper css-dev-only-do-not-override-os2fvf"
                >
                  <span
                    class="ant-checkbox ant-wave-target css-dev-only-do-not-override-os2fvf"
                  >
                    <input
                      class="ant-checkbox-input"
                      type="checkbox"
                    />
                    <span
                      class="ant-checkbox-inner"
                    />
                  </span>
                  <span
                    class="ant-checkbox-label"
                  >
                    Project B
                  </span>
                </label>
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            class="flex items-center min-h-6 mb-1"
            style="padding-left: 20px;"
          >
            <label
              class="ant-checkbox-wrapper css-dev-only-do-not-override-os2fvf"
            >
              <span
                class="ant-checkbox ant-wave-target css-dev-only-do-not-override-os2fvf"
              >
                <input
                  class="ant-checkbox-input"
                  type="checkbox"
                />
                <span
                  class="ant-checkbox-inner"
                />
              </span>
              <span
                class="ant-checkbox-label"
              >
                Reports
              </span>
            </label>
          </div>
          <div>
            <div>
              <div
                class="flex items-center min-h-6 mb-1"
                style="padding-left: 40px;"
              >
                <label
                  class="ant-checkbox-wrapper css-dev-only-do-not-override-os2fvf"
                >
                  <span
                    class="ant-checkbox ant-wave-target css-dev-only-do-not-override-os2fvf"
                  >
                    <input
                      class="ant-checkbox-input"
                      type="checkbox"
                    />
                    <span
                      class="ant-checkbox-inner"
                    />
                  </span>
                  <span
                    class="ant-checkbox-label"
                  >
                    Annual Report
                  </span>
                </label>
              </div>
            </div>
            <div>
              <div
                class="flex items-center min-h-6 mb-1"
                style="padding-left: 40px;"
              >
                <label
                  class="ant-checkbox-wrapper css-dev-only-do-not-override-os2fvf"
                >
                  <span
                    class="ant-checkbox ant-wave-target css-dev-only-do-not-override-os2fvf"
                  >
                    <input
                      class="ant-checkbox-input"
                      type="checkbox"
                    />
                    <span
                      class="ant-checkbox-inner"
                    />
                  </span>
                  <span
                    class="ant-checkbox-label"
                  >
                    Monthly Report
                  </span>
                </label>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        class="flex items-center min-h-6 mb-1"
        style="padding-left: 0px;"
      >
        <label
          class="ant-checkbox-wrapper css-dev-only-do-not-override-os2fvf"
        >
          <span
            class="ant-checkbox ant-wave-target css-dev-only-do-not-override-os2fvf"
          >
            <input
              class="ant-checkbox-input"
              type="checkbox"
            />
            <span
              class="ant-checkbox-inner"
            />
          </span>
          <span
            class="ant-checkbox-label"
          >
            Media
          </span>
        </label>
      </div>
      <div>
        <div>
          <div
            class="flex items-center min-h-6 mb-1"
            style="padding-left: 20px;"
          >
            <label
              class="ant-checkbox-wrapper css-dev-only-do-not-override-os2fvf"
            >
              <span
                class="ant-checkbox ant-wave-target css-dev-only-do-not-override-os2fvf"
              >
                <input
                  class="ant-checkbox-input"
                  type="checkbox"
                />
                <span
                  class="ant-checkbox-inner"
                />
              </span>
              <span
                class="ant-checkbox-label"
              >
                Images
              </span>
            </label>
          </div>
        </div>
        <div>
          <div
            class="flex items-center min-h-6 mb-1"
            style="padding-left: 20px;"
          >
            <label
              class="ant-checkbox-wrapper css-dev-only-do-not-override-os2fvf"
            >
              <span
                class="ant-checkbox ant-wave-target css-dev-only-do-not-override-os2fvf"
              >
                <input
                  class="ant-checkbox-input"
                  type="checkbox"
                />
                <span
                  class="ant-checkbox-inner"
                />
              </span>
              <span
                class="ant-checkbox-label"
              >
                Videos
              </span>
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`CheckboxTree should render disabled 1`] = `
<div>
  <div>
    <div>
      <div
        class="flex items-center min-h-6 mb-1"
        style="padding-left: 0px;"
      >
        <label
          class="ant-checkbox-wrapper ant-checkbox-wrapper-disabled css-dev-only-do-not-override-os2fvf"
        >
          <span
            class="ant-checkbox ant-wave-target css-dev-only-do-not-override-os2fvf ant-checkbox-disabled"
          >
            <input
              class="ant-checkbox-input"
              disabled=""
              type="checkbox"
            />
            <span
              class="ant-checkbox-inner"
            />
          </span>
          <span
            class="ant-checkbox-label"
          >
            Documents
          </span>
        </label>
      </div>
      <div>
        <div>
          <div
            class="flex items-center min-h-6 mb-1"
            style="padding-left: 20px;"
          >
            <label
              class="ant-checkbox-wrapper ant-checkbox-wrapper-disabled css-dev-only-do-not-override-os2fvf"
            >
              <span
                class="ant-checkbox ant-wave-target css-dev-only-do-not-override-os2fvf ant-checkbox-disabled"
              >
                <input
                  class="ant-checkbox-input"
                  disabled=""
                  type="checkbox"
                />
                <span
                  class="ant-checkbox-inner"
                />
              </span>
              <span
                class="ant-checkbox-label"
              >
                Projects
              </span>
            </label>
          </div>
          <div>
            <div>
              <div
                class="flex items-center min-h-6 mb-1"
                style="padding-left: 40px;"
              >
                <label
                  class="ant-checkbox-wrapper ant-checkbox-wrapper-disabled css-dev-only-do-not-override-os2fvf"
                >
                  <span
                    class="ant-checkbox ant-wave-target css-dev-only-do-not-override-os2fvf ant-checkbox-disabled"
                  >
                    <input
                      class="ant-checkbox-input"
                      disabled=""
                      type="checkbox"
                    />
                    <span
                      class="ant-checkbox-inner"
                    />
                  </span>
                  <span
                    class="ant-checkbox-label"
                  >
                    Project A
                  </span>
                </label>
              </div>
            </div>
            <div>
              <div
                class="flex items-center min-h-6 mb-1"
                style="padding-left: 40px;"
              >
                <label
                  class="ant-checkbox-wrapper ant-checkbox-wrapper-disabled css-dev-only-do-not-override-os2fvf"
                >
                  <span
                    class="ant-checkbox ant-wave-target css-dev-only-do-not-override-os2fvf ant-checkbox-disabled"
                  >
                    <input
                      class="ant-checkbox-input"
                      disabled=""
                      type="checkbox"
                    />
                    <span
                      class="ant-checkbox-inner"
                    />
                  </span>
                  <span
                    class="ant-checkbox-label"
                  >
                    Project B
                  </span>
                </label>
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            class="flex items-center min-h-6 mb-1"
            style="padding-left: 20px;"
          >
            <label
              class="ant-checkbox-wrapper ant-checkbox-wrapper-disabled css-dev-only-do-not-override-os2fvf"
            >
              <span
                class="ant-checkbox ant-wave-target css-dev-only-do-not-override-os2fvf ant-checkbox-disabled"
              >
                <input
                  class="ant-checkbox-input"
                  disabled=""
                  type="checkbox"
                />
                <span
                  class="ant-checkbox-inner"
                />
              </span>
              <span
                class="ant-checkbox-label"
              >
                Reports
              </span>
            </label>
          </div>
          <div>
            <div>
              <div
                class="flex items-center min-h-6 mb-1"
                style="padding-left: 40px;"
              >
                <label
                  class="ant-checkbox-wrapper ant-checkbox-wrapper-disabled css-dev-only-do-not-override-os2fvf"
                >
                  <span
                    class="ant-checkbox ant-wave-target css-dev-only-do-not-override-os2fvf ant-checkbox-disabled"
                  >
                    <input
                      class="ant-checkbox-input"
                      disabled=""
                      type="checkbox"
                    />
                    <span
                      class="ant-checkbox-inner"
                    />
                  </span>
                  <span
                    class="ant-checkbox-label"
                  >
                    Annual Report
                  </span>
                </label>
              </div>
            </div>
            <div>
              <div
                class="flex items-center min-h-6 mb-1"
                style="padding-left: 40px;"
              >
                <label
                  class="ant-checkbox-wrapper ant-checkbox-wrapper-disabled css-dev-only-do-not-override-os2fvf"
                >
                  <span
                    class="ant-checkbox ant-wave-target css-dev-only-do-not-override-os2fvf ant-checkbox-disabled"
                  >
                    <input
                      class="ant-checkbox-input"
                      disabled=""
                      type="checkbox"
                    />
                    <span
                      class="ant-checkbox-inner"
                    />
                  </span>
                  <span
                    class="ant-checkbox-label"
                  >
                    Monthly Report
                  </span>
                </label>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        class="flex items-center min-h-6 mb-1"
        style="padding-left: 0px;"
      >
        <label
          class="ant-checkbox-wrapper ant-checkbox-wrapper-disabled css-dev-only-do-not-override-os2fvf"
        >
          <span
            class="ant-checkbox ant-wave-target css-dev-only-do-not-override-os2fvf ant-checkbox-disabled"
          >
            <input
              class="ant-checkbox-input"
              disabled=""
              type="checkbox"
            />
            <span
              class="ant-checkbox-inner"
            />
          </span>
          <span
            class="ant-checkbox-label"
          >
            Media
          </span>
        </label>
      </div>
      <div>
        <div>
          <div
            class="flex items-center min-h-6 mb-1"
            style="padding-left: 20px;"
          >
            <label
              class="ant-checkbox-wrapper ant-checkbox-wrapper-disabled css-dev-only-do-not-override-os2fvf"
            >
              <span
                class="ant-checkbox ant-wave-target css-dev-only-do-not-override-os2fvf ant-checkbox-disabled"
              >
                <input
                  class="ant-checkbox-input"
                  disabled=""
                  type="checkbox"
                />
                <span
                  class="ant-checkbox-inner"
                />
              </span>
              <span
                class="ant-checkbox-label"
              >
                Images
              </span>
            </label>
          </div>
        </div>
        <div>
          <div
            class="flex items-center min-h-6 mb-1"
            style="padding-left: 20px;"
          >
            <label
              class="ant-checkbox-wrapper ant-checkbox-wrapper-disabled css-dev-only-do-not-override-os2fvf"
            >
              <span
                class="ant-checkbox ant-wave-target css-dev-only-do-not-override-os2fvf ant-checkbox-disabled"
              >
                <input
                  class="ant-checkbox-input"
                  disabled=""
                  type="checkbox"
                />
                <span
                  class="ant-checkbox-inner"
                />
              </span>
              <span
                class="ant-checkbox-label"
              >
                Videos
              </span>
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`CheckboxTree should render with some checked items 1`] = `
<div>
  <div>
    <div>
      <div
        class="flex items-center min-h-6 mb-1"
        style="padding-left: 0px;"
      >
        <label
          class="ant-checkbox-wrapper css-dev-only-do-not-override-os2fvf"
        >
          <span
            class="ant-checkbox ant-wave-target css-dev-only-do-not-override-os2fvf"
          >
            <input
              class="ant-checkbox-input"
              type="checkbox"
            />
            <span
              class="ant-checkbox-inner"
            />
          </span>
          <span
            class="ant-checkbox-label"
          >
            Documents
          </span>
        </label>
      </div>
      <div>
        <div>
          <div
            class="flex items-center min-h-6 mb-1"
            style="padding-left: 20px;"
          >
            <label
              class="ant-checkbox-wrapper css-dev-only-do-not-override-os2fvf"
            >
              <span
                class="ant-checkbox ant-wave-target css-dev-only-do-not-override-os2fvf"
              >
                <input
                  class="ant-checkbox-input"
                  type="checkbox"
                />
                <span
                  class="ant-checkbox-inner"
                />
              </span>
              <span
                class="ant-checkbox-label"
              >
                Projects
              </span>
            </label>
          </div>
          <div>
            <div>
              <div
                class="flex items-center min-h-6 mb-1"
                style="padding-left: 40px;"
              >
                <label
                  class="ant-checkbox-wrapper ant-checkbox-wrapper-checked css-dev-only-do-not-override-os2fvf"
                >
                  <span
                    class="ant-checkbox ant-wave-target css-dev-only-do-not-override-os2fvf ant-checkbox-checked"
                  >
                    <input
                      checked=""
                      class="ant-checkbox-input"
                      type="checkbox"
                    />
                    <span
                      class="ant-checkbox-inner"
                    />
                  </span>
                  <span
                    class="ant-checkbox-label"
                  >
                    Project A
                  </span>
                </label>
              </div>
            </div>
            <div>
              <div
                class="flex items-center min-h-6 mb-1"
                style="padding-left: 40px;"
              >
                <label
                  class="ant-checkbox-wrapper css-dev-only-do-not-override-os2fvf"
                >
                  <span
                    class="ant-checkbox ant-wave-target css-dev-only-do-not-override-os2fvf"
                  >
                    <input
                      class="ant-checkbox-input"
                      type="checkbox"
                    />
                    <span
                      class="ant-checkbox-inner"
                    />
                  </span>
                  <span
                    class="ant-checkbox-label"
                  >
                    Project B
                  </span>
                </label>
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            class="flex items-center min-h-6 mb-1"
            style="padding-left: 20px;"
          >
            <label
              class="ant-checkbox-wrapper css-dev-only-do-not-override-os2fvf"
            >
              <span
                class="ant-checkbox ant-wave-target css-dev-only-do-not-override-os2fvf"
              >
                <input
                  class="ant-checkbox-input"
                  type="checkbox"
                />
                <span
                  class="ant-checkbox-inner"
                />
              </span>
              <span
                class="ant-checkbox-label"
              >
                Reports
              </span>
            </label>
          </div>
          <div>
            <div>
              <div
                class="flex items-center min-h-6 mb-1"
                style="padding-left: 40px;"
              >
                <label
                  class="ant-checkbox-wrapper css-dev-only-do-not-override-os2fvf"
                >
                  <span
                    class="ant-checkbox ant-wave-target css-dev-only-do-not-override-os2fvf"
                  >
                    <input
                      class="ant-checkbox-input"
                      type="checkbox"
                    />
                    <span
                      class="ant-checkbox-inner"
                    />
                  </span>
                  <span
                    class="ant-checkbox-label"
                  >
                    Annual Report
                  </span>
                </label>
              </div>
            </div>
            <div>
              <div
                class="flex items-center min-h-6 mb-1"
                style="padding-left: 40px;"
              >
                <label
                  class="ant-checkbox-wrapper css-dev-only-do-not-override-os2fvf"
                >
                  <span
                    class="ant-checkbox ant-wave-target css-dev-only-do-not-override-os2fvf"
                  >
                    <input
                      class="ant-checkbox-input"
                      type="checkbox"
                    />
                    <span
                      class="ant-checkbox-inner"
                    />
                  </span>
                  <span
                    class="ant-checkbox-label"
                  >
                    Monthly Report
                  </span>
                </label>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        class="flex items-center min-h-6 mb-1"
        style="padding-left: 0px;"
      >
        <label
          class="ant-checkbox-wrapper css-dev-only-do-not-override-os2fvf"
        >
          <span
            class="ant-checkbox ant-wave-target css-dev-only-do-not-override-os2fvf"
          >
            <input
              class="ant-checkbox-input"
              type="checkbox"
            />
            <span
              class="ant-checkbox-inner"
            />
          </span>
          <span
            class="ant-checkbox-label"
          >
            Media
          </span>
        </label>
      </div>
      <div>
        <div>
          <div
            class="flex items-center min-h-6 mb-1"
            style="padding-left: 20px;"
          >
            <label
              class="ant-checkbox-wrapper ant-checkbox-wrapper-checked css-dev-only-do-not-override-os2fvf"
            >
              <span
                class="ant-checkbox ant-wave-target css-dev-only-do-not-override-os2fvf ant-checkbox-checked"
              >
                <input
                  checked=""
                  class="ant-checkbox-input"
                  type="checkbox"
                />
                <span
                  class="ant-checkbox-inner"
                />
              </span>
              <span
                class="ant-checkbox-label"
              >
                Images
              </span>
            </label>
          </div>
        </div>
        <div>
          <div
            class="flex items-center min-h-6 mb-1"
            style="padding-left: 20px;"
          >
            <label
              class="ant-checkbox-wrapper css-dev-only-do-not-override-os2fvf"
            >
              <span
                class="ant-checkbox ant-wave-target css-dev-only-do-not-override-os2fvf"
              >
                <input
                  class="ant-checkbox-input"
                  type="checkbox"
                />
                <span
                  class="ant-checkbox-inner"
                />
              </span>
              <span
                class="ant-checkbox-label"
              >
                Videos
              </span>
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
